src/services/healthcareProfessionalService.ts

```diff
@@ -324,7 +334,89 @@ function validateNames(
         })
     }
 
-    //TODO validate each name
+    if (names && names[0].firstName.length > 30) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (names && !names[0].firstName.trim()) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+    }
+
+    if (names && names[0].lastName.length > 30) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (names && !names[0].lastName.trim()) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.REQUIRED,
+            httpStatus: 400
+        })
+    }
+
+    if (names && names[0].middleName && names[0].middleName.length > 60) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'names',
+            errorCode: ErrorCode.INVALID_LENGTH_TOO_LONG,
+            httpStatus: 400
+        })
+    }
+
+    if (names) {
+        const checkForDigits = /\d+/g
+
+        const containsDigitsFirstName = checkForDigits.test(names[0].firstName)
+        const containsDigitsLastName = checkForDigits.test(names[0].lastName)
+        const containsDigitsMiddleName = names[0].middleName ? checkForDigits.test(names[0].middleName) : false
+
+        if (containsDigitsFirstName || containsDigitsLastName || containsDigitsMiddleName) {
+            validationResults.hasErrors = true
+            validationResults.errors?.push({
+                field: 'names',
+                errorCode: ErrorCode.INVALID_NAME,
+                httpStatus: 400
+            })
+        }
+    }
+
+    if (names && names[0].locale == 'en_US') {
```

Comment:
Maybe this as well is unnecessary, but I thought f en_US is used the name should not contain Javanese characters so we don't get a mix from two alphabets in a name. Again maybe this is excessive